In [42]:
import pandas as pd
from scipy.stats import chi2_contingency

## Bomb Plant per Map


Each round for all maps, a terrorist team may be able to plant the bomb. This test is done to see if the map being played upon affects the rate at which the terrorist are able to plant the bomb. To do this, the contingency table must be extracted; wherein the column indices are whether the bomb has been planted or not, and the row indices are the maps. This will then undergo chi-square test, since we are looking for the independence of two categorical variables.

With this in mind, the null hypothesis would be: "The map and the terrorist team being successful in planting a bomb is independent of each other."

The test will be at 95% confidence level, where p must be < 0.05 to reject the null hypothesis. 

In [43]:
main_df = pd.read_csv("demos.csv")

main_df.drop_duplicates(subset=['round_id'], inplace=True)
bomb_df = main_df[['round_id', 'map_name', 'round_first_site_hit', 'round_bomb_plant_site']]
bomb_df['bomb_planted'] = bomb_df['round_bomb_plant_site'].notnull().astype(int)


C:\Users\user\AppData\Local\Temp\ipykernel_22260\1264610674.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  bomb_df['bomb_planted'] = bomb_df['round_bomb_plant_site'].notnull().astype(int)


### Contingency table

In [44]:
bomb_plant_contingency_table = pd.crosstab(bomb_df['map_name'], bomb_df['bomb_planted'])
bomb_plant_contingency_table

bomb_planted    0    1
map_name              
de_ancient     55   79
de_anubis      38   45
de_dust2      141  155
de_inferno    120  165
de_mirage     150  143
de_nuke       143  119
de_train       35   40

### Chi-Square Test



In [45]:
chi2, p, dof, expected = chi2_contingency(bomb_plant_contingency_table)
print(f"p = {p:.4f}")

p = 0.0515


With p-value = 0.0515, we fail to reject the null hypothesis. Thus, it can be said that the terrorist team being able to successfully plant the bomb is independent with the map played. This can mean that the map does not necessarily favor or hinder terrorists in planting the bomb. There may be other factors that are more fit to be associated with the terrorist team being able to plant the bomb.

## Bomb Plant and First Site Hit

In some rounds, players do not plant at the same site as the first site that they entered in. There can be instances where players "rotate" to another site after they entered another site. This part of the testing determines the dependence of players planting the bomb in the same site they first entered in with the map being played upon. This will be tested via chi-square contingency as the relevant variables are categorical in nature.  

With this in mind, the null hypothesis would be: "The terrorist team successfully planting in the same site as they first entered in is independent with the map"

The test will be at 95% confidence level, where p must be < 0.05 to reject the null hypothesis. 

In [46]:
same_site_df = bomb_df.copy()
same_site_df.dropna(subset=['round_first_site_hit'], inplace=True)

same_site_df['same_site_hit'] = (same_site_df['round_first_site_hit'] == same_site_df['round_bomb_plant_site'])

same_site_df = same_site_df.loc[same_site_df['bomb_planted'] == 1]

### Contingency Table

In [47]:
same_site_contingency_table = pd.crosstab(same_site_df['map_name'], same_site_df['same_site_hit'])
same_site_contingency_table

same_site_hit  False  True 
map_name                   
de_ancient         7     46
de_anubis          5     25
de_dust2           7    112
de_inferno        12    104
de_mirage         10     74
de_nuke           13     73
de_train          11     20

### Chi-Square Test

In [48]:
chi2, p, dof, expected = chi2_contingency(same_site_contingency_table, correction=False)
print(f"p = {p:.4f}")

p = 0.0016


With p-value = 0.0016, we reject the null hypothesis. Thus, it can be said that having the bomb planted on the same site as the first site hit is dependent with map played. This means that certain maps may be associated with certain strategies with regards to attack "rotation". There can be maps where players often choose to rotate to another site, and there are maps where the first site hit is the optimal site to plant the bomb on. 